R/operators.R

```diff
@@ -1079,22 +1079,23 @@ path <- function(...) {
   } else if ("igraph.vertex" %in% class(e2)) {
     ## Adding vertices, possibly with attributes
     ## If there is a single unnamed argument, that contains the vertex names
-    wn <- which(names(e2) == "")
-    if (length(wn) == 1) {
-      names(e2)[wn] <- "name"
+    unnamed_elements_indices <- which(!rlang::have_name(e2))
+    if (length(unnamed_elements_indices) == 1) {
+      e2 <- rlang::set_names(e2[unnamed_elements_indices], "name")
```

Comment:
I guess the readability of this line is not obviously better but since I was using rlang above...